### ISSUE TITLE
Is there way to have multiple blueprint file?

### ISSUE THREAD

**Issue by azsec** [Azure/azure-blueprints#21]
Given the fact that a large blueprint requires a very long list of parameters in blueprint.json. Would it be ideal to have each artifact a blueprint file so we could manage easier? I think that could be technically done by PowerShell module.

**Comment by alex-frankel** [comment]
I'm not sure I'm following - today each artifact is in it's own file. Can you elaborate a bit?

**Comment by azsec** [comment]
Thank for your reply @alex-frankel . With the current design, there is one blueprint definition file that stores all parameters. Would it be an ideal approach if we could have each artifact a blueprint definition. For example I could manage my VM artifact's parameters in something called `blueprint.vm.json` ?

**Comment by alex-frankel** [comment]
I see - so instead of having a single blueprint.json, you want one for each artifact. Is that right?

If that's the case, this is not a change we are likely to make in the short term unfortunately.

**Comment by azsec** [comment]
Yes it is what I'd like - so hopefully we would have an easier way for managing large blueprint in the future. Thanks for the reply @alex-frankel

**Comment by tikicoder** [comment]
You can create multiple blueprints.  Also you could create a process for merging the blueprints. For me I use everything as yaml and convert to json and build out the structure. I have the yaml broken out into groups.

**Comment by torreymicrosoft** [comment]
Thanks for the enhancement request to split a large blueprint into multiple files for easier management.

Heads-up on some important news: Azure Blueprints is being retired on January 31, 2027, with a phased wind-down starting July 31, 2026. After retirement the API, Azure CLI, PowerShell, and portal experience go away, and any definitions, versions, or assignments you have not exported are deleted (resources already deployed are not removed). Because of this, we are no longer shipping fixes or new features for Azure Blueprints.

The recommended replacement is Azure Deployment Stacks, which provides resource grouping, full lifecycle management, and deny assignment (lock) protection equivalent to blueprint locks. You can also publish your definitions as Template Specs or store them in Git for versioning.

Full details and the phased timeline are here:
- Announcement: https://aka.ms/AzureBlueprintsRetirementAnnouncement
- Migration guidance: https://aka.ms/AzureBlueprintsMigration
- FAQ: https://aka.ms/AzureBlueprintsRetirementFAQ

We are closing this issue as part of the Azure Blueprints retirement. If you run into anything while moving to Deployment Stacks or Template Specs, please open an issue in that repository so the right team can help. Thanks for using Azure Blueprints.
